### PR TITLE
feat(historical-view): add mockup

### PR DESCRIPTION
**Pull request by RR-Ardhika**
Sample view image:

image

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,113 +1,56 @@
...
-        </a>
-      </div>
-    </main>
+export default function Home() {
+  return (
+    <div className="flex flex-col mt-[72px] pt-4 px-4">
+      {juzCard()}
+      {surahCard()}
+      {ayahCard()}
+      {repeat([juzCard(), surahCard(), ayahCard()], 7)}
```
Comment (taufiqelrahman) [inline src/app/page.tsx:53]: is this for dummy purposes?

File: src/app/components/navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default function Navbar() {
+  return (
+    <div className="flex mt-auto sticky bottom-0">
```
Comment (taufiqelrahman) [inline src/app/components/navbar.tsx:3]: let's use semantic html tags
```suggestion
    <nav className="flex mt-auto sticky bottom-0">
```

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,113 +1,56 @@
-import Image from "next/image";
+const classCardDiv = 'p-4 mb-5 bg-teal text-white rounded-lg';
+const classCardTitle = 'text-xl font-black';
+const classCardData = 'font-normal';
+const classCardDate = 'font-extralight';
```
Comment (taufiqelrahman) [inline src/app/page.tsx:4]: maybe group it as object?
```suggestion
const cardClassnames = {
  container: 'p-4 mb-5 bg-teal text-white rounded-lg',
  title: 'text-xl font-black',
  data: 'font-normal',
  date: 'font-extralight',
}
```

File: src/app/components/header.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default function Header() {
+  return (
+    <h1 className="fixed w-full p-4 bg-teal text-4xl text-white">Murojaah</h1>
```
Comment (taufiqelrahman) [inline src/app/components/header.tsx:3]: let's use semantic html tags
```suggestion
    <header className="fixed w-full p-4 bg-teal text-4xl text-white">Murojaah</h1>
```

File: src/app/page.tsx (hunk 1 above)
Comment (RR-Ardhika) [reply src/app/page.tsx:53]: yes, later will change to retrieve data from api

File: src/app/components/navbar.tsx (hunk 2 above)
Comment (RR-Ardhika) [reply src/app/components/navbar.tsx]: already updated on https://github.com/RR-Ardhika/murojaah/pull/3/commits/03e94125dce8075c8df7a729973197ff0498dae9

File: src/app/components/header.tsx (hunk 4 above)
Comment (RR-Ardhika) [reply src/app/components/header.tsx]: already updated on https://github.com/RR-Ardhika/murojaah/pull/3/commits/03e94125dce8075c8df7a729973197ff0498dae9

File: src/app/page.tsx (hunk 3 above)
Comment (RR-Ardhika) [reply src/app/page.tsx]: already updated on https://github.com/RR-Ardhika/murojaah/pull/3/commits/201f7d6c9f8f8a8ef68bde319a412d68490fa2a8

File: src/app/page.tsx [hunk 5]
```diff
@@ -1,113 +1,58 @@
...
-              -&gt;
-            </span>
-          </h2>
-          <p className={`m-0 max-w-[30ch] text-sm opacity-50`}>
-            Find in-depth information about Next.js features and API.
-          </p>
-        </a>
+function juzCard() {
+  return (
+    <div className={cardClassnames.container}>
```
Comment (taufiqelrahman) [inline src/app/page.tsx:17]: can we create a reusable `Card` component? maybe with props like this:
```
title: string;
data: string[];
date: string;
```

Comment (RR-Ardhika) [reply src/app/page.tsx]: already updated on https://github.com/RR-Ardhika/murojaah/pull/3/commits/8ea0113c21753aca50bdc14f0e83e22d5dc1c9af

File: src/app/components/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+function juzCard(): JSX.Element {
+  return (
+    <div className={cardClassnames.container}>
+      <p className={cardClassnames.title}>Juz 28</p>
+      <p className={cardClassnames.data}>Murojaah by using Memory</p>
+      <p className={cardClassnames.data}>Total Murojaah is 100</p>
+      <p className={cardClassnames.date}>Sat, Feb 10 &apos;24</p>
+    </div>
+  );
+}
```
Comment (taufiqelrahman) [inline src/app/components/Card.tsx:38]: this is what i had in mind for reusabled `Card`
```
const Card = ({title, contents, date}: Props) => (
    <div className={cardClassnames.container}>
      <p className={cardClassnames.title}>{title}</p>
      {contents.map(content => (
        <p className={cardClassnames.data}>{content}</p>
      )}
      <p className={cardClassnames.date}>{date}</p>
    </div>
)
```
and then in page.tsx we can create these:
```
const JuzCard = () => (
  <Card
    title="Juz 28"
    contents={["Murojaah by using Memory", "Total Murojaah is 100"]}
    date="Sat, Feb 10 &apos;24"
    />
)

...
const SurahCard ...
const AyahCard ...
```

File: src/app/page.tsx [hunk 7]
```diff
@@ -1,58 +1,26 @@
...
-      <p className={cardClassnames.data}>Surah 1 Al-Baqarah</p>
-      <p className={cardClassnames.data}>Murojaah by using Memory</p>
-      <p className={cardClassnames.data}>Total Murojaah is 100</p>
-      <p className={cardClassnames.date}>Sat, Feb 10 &apos;24</p>
-    </div>
-  );
-}
+export default function Home(): JSX.Element {
+  const cards: JSX.Element[] = [
+    Card({ cardType: 0 }),
```
Comment (taufiqelrahman) [inline src/app/page.tsx:13]: fyi you can also call it this way for easier read:
```
const cards: JSX.Element[] = [
    <Card cardType={0} />,
]
```

Comment (RR-Ardhika) [reply src/app/page.tsx]: already updated on commit https://github.com/RR-Ardhika/murojaah/pull/3/commits/801582a821f492cd261221ccdb993e436a76482d

File: src/app/components/Card.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  container: 'p-4 mb-5 bg-teal text-white rounded-lg',
+  title: 'text-xl font-black',
+  data: 'font-normal',
+  date: 'font-extralight',
+};
+
+export default function Card({ cardType }: Props): JSX.Element {
+  let el: JSX.Element = <></>;
+  switch (cardType) {
+    case 0:
```
[the remark this answers is not in the document]
Comment (RR-Ardhika) [reply src/app/components/Card.tsx]: already updated on https://github.com/RR-Ardhika/murojaah/pull/3/commits/9845341fe9935b6379d220f72bc78bbccef5e85b

File: src/app/components/Card.tsx (hunk 6 above)
Comment (RR-Ardhika) [reply src/app/components/Card.tsx]: already updated on https://github.com/RR-Ardhika/murojaah/pull/3/commits/69019b3f97b3a880a20250ac7d32f9eb7385a0dc

File: src/app/page.tsx [hunk 9]
```diff
@@ -1,18 +1,10 @@
...
-
-  return items;
-}
-
 export default function Home(): JSX.Element {
   const cards: JSX.Element[] = [
-    Card({ cardType: 0 }),
-    Card({ cardType: 1 }),
-    Card({ cardType: 2 }),
+    <Card key={Math.floor(Math.random())} cardType={0} />,
```
Comment (taufiqelrahman) [inline src/app/page.tsx]: `Math.floor(Math.random())`
wont this recalculate every rerender?

File: src/app/components/Card.tsx [hunk 10]
```diff
@@ -11,53 +19,53 @@ const cardClassnames: Record<string, string> = {
...
+      <p className={cardClassnames.title}>Surah {props.surah} {props.surahName}</p>
+      <p className={cardClassnames.data}>Murojaah by using {props.murojaahMethod}</p>
+      <p className={cardClassnames.data}>Total Murojaah is {props.totalMurojaah}</p>
+      <p className={cardClassnames.date}>{props.occuredAt}</p>
     </div>
   );
 }
 
-function ayahCard(): JSX.Element {
+function ayahCard(props: Props): JSX.Element {
```
Comment (taufiqelrahman) [inline src/app/components/Card.tsx:71]: how about differentiating props for each ayah, surah, juz?

```
interface JuzProps {
...
}

interface SurahProps extends JuzProps {
...
}

interface AyahProps extends SurahProps {
...
}
```

File: src/app/components/Card.tsx [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+      break;
+    case CardType.Surah:
+      el = surahCard(props);
+      break;
+    case CardType.Ayah:
+      el = ayahCard(props);
+      break;
+  }
+
+  return el;
```
Comment (taufiqelrahman) [inline src/app/components/Card.tsx:36]:
```suggestion
  switch (props.cardType) {
    case CardType.Juz:
      return juzCard(props);
    case CardType.Surah:
      return surahCard(props);
    case CardType.Ayah:
      return ayahCard(props);
    default:
      return <></>;
  }
```

File: src/app/components/Card.tsx [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+function juzCard(props: Props): JSX.Element {
```
Comment (taufiqelrahman) [inline src/app/components/Card.tsx:39]: if this is a function, lets rename it as verb like `renderJuzCard`. but if this is component, then lets use `CamelCase` and define in `const xxx = () => ()`

File: src/app/page.tsx (hunk 9 above)
Comment (RR-Ardhika) [reply src/app/page.tsx]: when use Card({ ... }) the eslint not raise error about the key prop, but using component tag directly it need us to use the key prop

also later probably the key value will retrieved from record's id so wont recalculate on each render, should we use the Card() again or continue to use component tag? based on convention array values of JSX.Element[] type

File: src/app/components/Card.tsx (hunk 11 above)
Comment (RR-Ardhika) [reply src/app/components/Card.tsx]: updated on https://github.com/RR-Ardhika/murojaah/pull/3/commits/35abc98c2811a9680b68a0e3d5b8b054c9f9536a

File: src/app/page.tsx (hunk 9 above)
Comment (RR-Ardhika) [reply src/app/page.tsx]: after see the code again I realize this only for mockup data, so I added todo comment on https://github.com/RR-Ardhika/murojaah/pull/3/commits/494ab4d2197299e82a6bfe2ae85cf1a721b31d52

**Pull request merged**